provider/sgx.py

```diff
@@ -0,0 +1,200 @@
+"""
+Module for sgx relevant operations.
+"""
+import logging
+import re
+
+from avocado.utils import process
+from virttest.staging import utils_memory
+from virttest.utils_misc import normalize_data_size
+
+LOG_JOB = logging.getLogger('avocado.test')
+
+
+class SGXError(Exception):
+    """ General SGX error"""
+    pass
+
+
+def _get_epc_size(output):
+    """
+    Get epc size from system message
+
+    :param output: epc section info
+    :return: total epc size
+    """
+    epc_size = 0
+    for line in output.splitlines():
+        tmp_epc = re.search(r'\b0x[0-9a-fA-F]*-0x[0-9a-fA-F]*\b',
+                            str(line))
+        if tmp_epc:
+            epc_size += int(tmp_epc[0].split('-')[1], 16) - int(
+                tmp_epc[0].split("-")[0], 16) + 1
+    return epc_size
+
+
+class SGXHostCapability(object):
+    """
+    Hypervisor sgx capabilities check.
+    """
+
+    def __init__(self, test, params):
+        """
+
+        :param test: Context of test.
+        :param params: params of running ENV.
+        """
+        self._test = test
+        self._params = params
+        self.host_epc_size = 0
+
+    def valid_sgx_cap(self):
+        """
+        Valid if host enable sgx
+
+        :return: True/False
+        """
+        try:
+            host_sgx_msg = process.system_output(
+                "journalctl|grep -i 'sgx: EPC section'", shell=True)
+        except Exception as e:
+            self._test.cancel("Host sgx capability check fail %s" % e)
+        else:
+            self.host_epc_size = _get_epc_size(host_sgx_msg)
+
+    def valid_host_numa_node(self):
+        """
+        Valid if host numa nodes satisfy test requirement
+
+        :return: True/False
```

Comment:
pls refer to the comments above

Reply:
can we change the name to validate_numa_node_count? It looks we only check the count in the function

Reply:
Thanks make sense to me